src/p4p/gw.py

```diff
@@ -720,7 +744,10 @@ def run(self):
     def sleep(dly):
         time.sleep(dly)
 
-def main(args=None):
+args = None
+
+def main():
+    global args
```

Comment:
No globals please!  Consider the unittests.  Having a module level attribute is no better (and having both is redundant).